pkg/commands/compute/deploy_test.go

```diff
@@ -2239,6 +2240,95 @@ func TestDeploy(t *testing.T) {
 	}
 }
 
+func TestDeploy_ActivateBeacon(t *testing.T) {
+	//
+	// We're going to chdir to a deploy environment,
+	// so save the PWD to return to, afterwards.
+	pwd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create test environment
+	rootdir := testutil.NewEnv(testutil.EnvOpts{
+		T: t,
+		Copy: []testutil.FileIO{
+			{
+				Src: filepath.Join("testdata", "deploy", "pkg", "package.tar.gz"),
+				Dst: filepath.Join("pkg", "package.tar.gz"),
+			},
+		},
+		Write: []testutil.FileIO{
+			{
+				Src: "This is my data for the KV Store 'store_one' baz field.",
+				Dst: "kv_store_one_baz.txt",
+			},
+		},
+	})
+	defer os.RemoveAll(rootdir)
+
+	// Before running the test, chdir into the build environment.
+	// When we're done, chdir back to our original location.
+	// This is so we can reliably copy the testdata/ fixtures.
+	if err := os.Chdir(rootdir); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		_ = os.Chdir(pwd)
+	}()
+
+	stdout := threadsafe.Buffer{}
+	args := testutil.Args("compute deploy --auto-yes --non-interactive")
+	recordingHTTP := testutil.NewRecordingHTTPClient()
+	recordingHTTP.SingleResponse(&http.Response{
+		StatusCode: http.StatusNoContent,
+		Status:     http.StatusText(http.StatusNoContent),
+		Body:       io.NopCloser(bytes.NewBufferString("")),
+	}, nil)
+	fmt.Printf("recording HTTP with...\n%#v\n", recordingHTTP)
```

Comment:
Could we have this encapsulated inside of the `NewRecordingHTTPClient`.

Reply:
Oops, that was left over from some debugging